src/override/malloc.cc

```diff
@@ -116,6 +116,36 @@ extern "C"
   }
 #endif
 
+#if !defined(__NetBSD__)
+  SNMALLOC_EXPORT int
+    SNMALLOC_NAME_MANGLE(reallocarr)(void* ptr, size_t nmemb, size_t size)
+  {
+    bool overflow = false;
+    auto& a = ThreadAlloc::get();
+    size_t sz = bits::umul(size, nmemb, overflow);
+    if (SNMALLOC_LIKELY(sz == 0))
+    {
+      return 0;
+    }
+    if (overflow)
+    {
+      return EOVERFLOW;
+    }
+    size_t osz = a.alloc_size(ptr);
+    void* p = a.alloc(osz);
+    memcpy(p, ptr, osz);
```

Comment:
This copy seems dubious?  Does this corrupt the stack in the common usage?
* `p` - is a pointer to an allocation
* `ptr` - is a pointer to a pointer to an allocation.